adapter/internal/oasparser/model/mgw_common.go

```diff
@@ -0,0 +1,196 @@
+func ResolveThrottlingTier(vendorExtensions map[string]interface{}) string {
+	xTier := ""
+	if x, found := vendorExtensions[constants.XWso2ThrottlingTier]; found {
+		if val, ok := x.(string); ok {
+			xTier = val
+		}
+	} else if y, found := vendorExtensions[constants.XThrottlingTier]; found {
+		if val, ok := y.(string); ok {
+			xTier = val
+		}
+	}
+	return xTier
+}
+
+// ResolveDisableSecurity extracts the value of x-auth-type extension.
+// if the property is not available, false is returned.
+// If the API definition is fed from API manager, then API definition contains
+// x-auth-type as "None" for non secured APIs. Then the return value would be true.
+// If the API definition is fed through apictl, the users can use either
+// x-wso2-disable-security : true/false to enable and disable security.
+func ResolveDisableSecurity(vendorExtensions map[string]interface{}) bool {
+	disableSecurity := false
+	y, vExtAuthType := vendorExtensions[constants.XAuthType]
+	z, vExtDisableSecurity := vendorExtensions[constants.XWso2DisableSecurity]
+	if vExtDisableSecurity {
+		// If x-wso2-disable-security is present, then disableSecurity = val
+		if val, ok := z.(bool); ok {
+			disableSecurity = val
+		}
+	}
+	if vExtAuthType && !disableSecurity {
+		// If APIs are published through APIM, all resource levels contains x-auth-type
+		// vendor extension.
+		if val, ok := y.(string); ok {
+			// If the x-auth-type vendor ext is None, then the API/resource is considerd
+			// to be non secure
+			if val == constants.None {
+				disableSecurity = true
+			}
+		}
+	}
+	return disableSecurity
+}
+
+func getHTTPEndpoint(rawURL string) (*Endpoint, error) {
+	return getHostandBasepathandPort(constants.HTTP, rawURL)
+}
+
+func getWebSocketEndpoint(rawURL string) (*Endpoint, error) {
+	return getHostandBasepathandPort(constants.WS, rawURL)
+}
+
+func getHostandBasepathandPort(apiType string, rawURL string) (*Endpoint, error) {
+	var (
+		basepath string
+		host     string
+		port     uint32
+		urlType  string
+	)
+
+	if !strings.Contains(rawURL, "://") {
+		if apiType == constants.HTTP {
+			rawURL = "http://" + rawURL
+		} else if apiType == constants.WS {
+			rawURL = "ws://" + rawURL
+		}
+	}
+
+	parsedURL, err := url.Parse(rawURL)
+	if err != nil {
+		logger.LoggerOasparser.Errorf("Failed to parse the malformed endpoint %v. Error message: %v", rawURL, err)
+		return nil, err
+	}
+
+	// Hostname validation
+	if err == nil && !regexp.MustCompile(hostNameValidator).MatchString(parsedURL.Hostname()) {
+		logger.LoggerOasparser.Error("Malformed endpoint detected (Invalid host name) : ", rawURL)
+		return nil, errors.New("malformed endpoint detected (Invalid host name) : " + rawURL)
+	}
+
+	host = parsedURL.Hostname()
+	basepath = parsedURL.Path
+	if parsedURL.Port() != "" {
+		u32, err := strconv.ParseUint(parsedURL.Port(), 10, 32)
+		if err != nil {
+			logger.LoggerOasparser.Error("Error passing port value to mgwSwagger", err)
```

Comment:
passing port value to mgwswagger dosen't have a meaning. lets change it. 
ex: `Endpoint port is not in the expected format.`

Reply:
Fixed with https://github.com/wso2/product-microgateway/pull/2635/commits/58fe8546c29ba231eca3172df82e118d9194cbd8